src/rt64lib/shaders/Lights.hlsli

```diff
@@ -95,27 +133,50 @@ float3 ComputeLight(uint2 launchIndex, uint lightIndex, float3 rayDirection, uin
 		float sampleIntensityFactor = pow(max(1.0f - (sampleDistance / lightRadius), 0.0f), lightAttenuation);
 		float3 reflectedLight = reflect(-sampleDirection, normal);
 		float NdotL = max(dot(normal, sampleDirection), 0.0f);
-		float sampleLambertFactor = lerp(NdotL, 1.0f, ignoreNormalFactor) * sampleIntensityFactor;
-		float sampleShadowFactor = 1.0f;
+        float lerpedNdotL = lerp(NdotL, 1.0f, ignoreNormalFactor);
+        float sampleLambertFactor = lerpedNdotL * sampleIntensityFactor;
+        float2 sampleSpecularFactor = float2(0, 0);
+        // TODO: Instead of a global flag, make this material driven
+        if (processingFlags & 0x8) {
+            sampleSpecularFactor = pow(CalculateSpecularity(normal, position, viewPosition, samplePosition, (roughness * .9 + .1), fresnelFactor) * lerpedNdotL * sampleIntensityFactor, max(specularExponent, EPSILON));
+        } else {
+            sampleSpecularFactor.x = pow(max(saturate(dot(reflectedLight, -rayDirection) * sampleIntensityFactor), 0.0f), specularExponent);
+        }
+        float sampleShadowFactor = 1.0f;
 		if (checkShadows) {
 			sampleShadowFactor = TraceShadow(position, sampleDirection, RAY_MIN_DISTANCE + shadowRayBias, (sampleDistance - shadowOffset));
-		}
-
-		float3 sampleSpecularityFactor = specular * pow(max(saturate(dot(reflectedLight, -rayDirection) * sampleIntensityFactor), 0.0f), specularExponent);
-		lLambertFactor += sampleLambertFactor / maxSamples;
-		lSpecularityFactor += sampleSpecularityFactor / maxSamples;
+        }
+	
+        float ks = 1.0;
+        float kd = 1.0;
+        float pi = 1.0;
+        if (processingFlags & 0x8)
```

Comment:
Will need more detail on what this flag (constant name or change to a boolean) is to understand what it's doing.